src/frontends/tensorflow_common/src/op/bincount.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "common_op_table.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/unsqueeze.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_bincount_op(const NodeContext& node) {
+    default_op_checks(node, 3, {"Bincount"});
+    auto arr = node.get_input(0);
+    auto size = node.get_input(1);
+    auto weights = node.get_input(2);
+
+    // arr: A Tensor of type int32. int32 Tensor.
+    auto arr_type = arr.get_element_type();
+    TENSORFLOW_OP_VALIDATION(node, arr_type == element::i32, "arr type is not supported: ", arr_type);
+
+    // size: A Tensor of type int32. non-negative int32 scalar Tensor.
+    auto size_type = size.get_element_type();
+    TENSORFLOW_OP_VALIDATION(node, size_type == element::i32, "size type is not supported: ", arr_type);
+    TENSORFLOW_OP_VALIDATION(node, is_scalar(size.get_shape()), "size must be a scalar");
```

Comment:
I think we should not check types of `arr` and `size`. We should assume that types are correct and follows the specification: https://www.tensorflow.org/api_docs/python/tf/raw_ops/Bincount.
Otherwise, the model is inconsistent so that TF don't even support it.
So please remove these checks

Reply:
got it